.chloggen/bugfix_target-info-required-attrs.yaml

```
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: prometheusremotewrite
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Change prometheusremotewrite.FromMetrics so that the target_info metric is only generated if either of the OTel resource attributes service.name and service.instance.id is defined.
```

```suggestion
note: Change prometheusremotewrite.FromMetrics so that the target_info metric is only generated if either of the OTel resource attributes service.name or service.instance.id is defined.
```

Otherwise it sounds like both need to be defined.

---

I'm not sure, to me it sounds correct, because you say either of a certain group (which is made up of those two). Either of the OTel resource attributes service.name or service.instance makes it sound as if the group is ill defined, i.e. it could either consist of service.name or of service.instance.id.

---

I personally think for "if" to make sense, it should be phrased "if either the OTel resource attribute service.name or the OTel resource attribute service.instance.id is defined"

---

Should have mention, this is just a NIT, since on my first reading I thought you need both defined otherwise you won't get the target_info metric.

---

Thanks :) I can also rewrite the sentence so it's clearer.

---

Tweaked the sentence.